engine-rest/engine-rest-openapi/README.md

```diff
@@ -0,0 +1,146 @@
+* for the `property` macros DO NOT forget to put `last = true` param for the last property, that will take care for the comas in the json file
+* the dto that have sorting or pagination properties should use the [common templates](#commons).
+
+### paths
+
+Contains the endpoints definitions of the Rest API. Instructions:
+* each resource has its own folder under /paths (e.g. `/process-instance`, `/deployment`)
+* each resource directory contains `all.ftl` file that contains endpoint paths, methods and includes endpoint definition templates.
+* the path of the endpoint is resolved to a folder structure (e.g. get process instance count `GET /process-instance/count` goes to `/paths/process-instance/count`).
+* the dinamic endpoints should be structured with brakes like `process-instance/{id}/variables/{varName}/data`, then the path parameters (`id` and `varName`) should be always included in the endpoint definition and marked as `required`
+* the name of the method's request (GET, POST, PUT, DELETE, OPTIONS) is the name of the template file (get.ftl, post.frl, etc.).
+* each endpoint definition has unique `operationId` that will be used for the generation of clients. Most of the cases the java method name should be used (e.g. `deleteProcessInstancesAsync`). When this is not possible please define it according to the Java conventions.
+* each endpoint definition contains tag of its resource (e.g. `Process instance`, `Deployment`).
+* each endpoint definition contains description.
+* each endpoint definition contains at least one response defined.
+* use the macros from the previous section when possible
+* for the `property` and `param` macros DO NOT forget to put last = true param for the last property/parameter, that will take care for the comas in the json file
+* the endpoints that have sorting or pagination properties/parameter should use the [common templates](#commons).
+
+#### commons
+
+Contains common templates that can be reused when it's possible.
+
+##### sorting
+
+* [sort-params.ftl](./src/main/templates/paths/commons/sort-params.ftl)
+* [sort-props.ftl](./src/main/templates/paths/commons/sort-props.ftl)
+
+Please set the `sortByValues` enumeration whenever the template is in use and do not forget to assign the `last = true` if this is the last parameter/property (taking care for the commas in the json):
+```
+<#-- <#assign last = true >  --> <#-- remove comment if last param  -->
+<#assign sortByValues = ['"instanceId"', '"definitionId"', '"definitionKey"', '"definitionName"', '"definitionVersion"', '"businessKey"', '"startTime"', '"endTime"', '"duration"', '"tenantId"']>
+<#include "/paths/commons/sort-props.ftl" >
+```
+
+##### pagination
+
+* [pagination-params.ftl](./src/main/templates/paths/commons/pagination-params.ftl)
+
+Use whenever `firstResult` and `maxResults` are part of the endpoint parameters. Do not forget to assigne `last=true` param in case those are the last parameters:
```

Comment:
```suggestion
Use whenever `firstResult` and `maxResults` are part of the endpoint parameters. Do not forget to assign `last=true` param in case those are the last parameters:
```